mne/viz/evoked.py

```diff
@@ -1191,3 +1191,342 @@ def plot_evoked_joint(evoked, times="peaks", title='', picks=None,
+            ymin = 0
+
+    # deal with dict/list of lists, and calculate the CI
+    if not isinstance(evokeds[conditions[0]], Evoked):
+        from ..evoked import combine_evoked
+        if not isinstance(evokeds[conditions[0]][0], Evoked):
+            raise ValueError("evokeds must be an `mne.Evoked` "
+                             "or of a collection of `mne.Evoked`s")
+
+        if ci and picks is not None:
+            # calculate the CI
+            sem_array = {}
+            for condition in conditions:
+                # this will fail if evokeds do not have the same structure
+                # (e.g. channel count)
+                if ch_type == 'grad' and picks is not None:
+                    data = np.asarray([
+                        _merge_grad_data(
+                            evoked_.data[pairpicks, :]).mean(0)[picks, :]
+                        for evoked_ in evokeds[condition]])
+                else:
+                    data = np.asarray([evoked_.data[picks, :].mean(0)
+                                       for evoked_ in evokeds[condition]])
+                sem_array[condition] = _ci(data, ci)
+
+        # get the grand mean
+        for cond in conditions:
+            evokeds[cond] = combine_evoked(evokeds[cond], weights='equal')
+
+        if picks is None:
+            warn("CI not drawn if plotting GFP.")
+
+    else:
+        ci = False
+
+    # let's plot!
+    if ax is None:
+        fig, ax = plt.subplots(1, 1)
+        fig.set_size_inches(8, 6)
+
+    # style the individual condition time series
+    if not isinstance(colors, dict):
+        colors_ = ['#e41a1c', '#377eb8', '#4daf4a', '#984ea3', '#ff7f00',
+                   '#1b9e77', '#d95f02', '#7570b3', '#e7298a', '#66a61e']
+        colors = dict((condition, color) for condition, color
+                      in zip(conditions, colors_))
+    else:
+        style_error = "Condition {} could not be mapped to a color."
+        for condition in conditions:
+            if condition not in colors:
+                if "/" not in condition:
+                    raise ValueError(style_error.format(condition))
+                for color in colors:
+                    if color in condition.split("/"):
+                        colors[condition] = colors[color]
+                        break
+
+    if not isinstance(linestyles, dict):
+        linestyles = dict((condition, linestyle) for condition, linestyle in
+                          zip(conditions, ['-'] * len(conditions)))
+    else:
+        style_error = "Condition {} could not be mapped to a linestyle."
+        for condition in conditions:
+            if condition not in linestyles:
+                if "/" not in condition:
+                    raise ValueError(style_error.format(condition))
+                for linestyle in linestyles:
+                    if linestyle in condition.split("/"):
+                        linestyles[condition] = linestyles[linestyle]
+                        break
```

Comment:
this loop in a loop is almost the same as checking for colors above - maybe reduce to one little function?